src/structs/types/cache_information.rs

```diff
@@ -147,6 +155,37 @@ impl Serialize for SMBiosCacheInformation<'_> {
     }
 }
 
+/// # Maximum memory capacity, in kilobytes, for this cache item
+#[derive(Serialize, Debug, PartialEq, Eq)]
+pub enum CacheMemorySize {
+    /// Maximum memory capacity in Kilobytes
+    Kilobytes(u64),
+    /// Use `maximum_cache_size_2` to retrieve the maximum capacity
+    SeeMaximumCacheSize2,
+}
+
+impl From<u16> for CacheMemorySize {
+    fn from(raw: u16) -> Self {
+        match raw {
+            0xFFFF => CacheMemorySize::SeeMaximumCacheSize2,
+            // When bit 15 is set, the size units of the remaining bits are 64k
+            _ if raw & 0x8000 == 0x8000 => CacheMemorySize::Kilobytes((raw & 0x7FFF) as u64 * 64),
+            _ => CacheMemorySize::Kilobytes(raw as u64),
+        }
+    }
+}
+
+impl From<u32> for CacheMemorySize {
```

Comment:
Maximum Cache Size 2 and Install Size 2 are DWORD (u32) and follow the same logic:

If this field is present, for cache sizes of
2047 MB or smaller the value in the Max
size in given granularity portion of the field
equals the size given in the corresponding
portion of the Maximum Cache Size field,
and the Granularity bit matches the value of
the Granularity bit in the Maximum Cache
Size field.
For Cache sizes greater than 2047 MB, the
Maximum Cache Size field is set to 0xFFFF
and the Maximum Cache Size 2 field is
present, the Granularity bit is set to 1b, and
the size set as required; see 7.8.1.
Bit 31 Granularity
0 – 1K granularity
1 – 64K granularity (always 1b
for cache sizes >2047 MB)
Bits 30:0 Max size in given granularity